### ISSUE TITLE
Is this a potential bug?

### ISSUE THREAD

**Issue by pengzhangzhi** [aqlaboratory/openfold#179]
Note that `expand` does not copy the data. Instead, it returns a new view of the self tensor.
https://pytorch.org/docs/stable/generated/torch.Tensor.expand.html#torch.Tensor.expand

 Do you think this is a potential bug? All the resulting rotations are the same in batch dimension, they share the  same contents in memory. If one rotation is updated by backprop or something, all the rotations would be changed, even though the rotation that belongs to other amino acids or proteins .

https://github.com/aqlaboratory/openfold/blob/cd41d73f31af7555e787eed134a302805c40502a/openfold/utils/rigid_utils.py#L98

**Comment by gahdritz** [comment]
You're right, but I don't really see it being an issue. Any future operation that differs across these singleton batch dimensions is going to cause the tensor to be broadcasted. As a sanity measure, I've added a `.contiguous()` call here in 86613a9, but I'm quite certain it's not necessary.

**Comment by pengzhangzhi** [comment]
Great! I have more question. What if the batch dim is not singleton? Then the tensors would not be broadcasted and the `rots` suffers from the aforementioned issue? Am I right?

**Comment by gahdritz** [comment]
No, that should still be fine. As soon as you try to rotate a batched tensor with a non-singleton batch dimension, the rotations will expand to match.
